packages/lune/src/builder.ts

```diff
@@ -52,30 +24,11 @@ export async function buildPlugin(path: string, to: string, cfg: LuneCfg, minify
 
   if (!entryPoint) throw new Error("failed to find entrypoint - check your working directory and config");
 
-  await build({
-    entryPoints: [entryPoint],
-    outfile,
-    bundle: true,
-    minify,
-    plugins: [
-      ...(cfg.prePlugins ?? []),
-      solidPlugin(),
-      (cfg.cssModules
-        ? sassPlugin({
-            style: "compressed",
-            sourceMap: false,
-            transform: postcssModules({
-              localsConvention: "camelCaseOnly",
-              inject: cfg.cssModules === "legacy" ? false : "shelter",
-            } as any),
-          })
-        : sassPlugin({ style: "compressed", sourceMap: false, type: "css-text" })) as any, // bad but version conflicts suck
-      shelterEsbuildResolver(),
-      ...(cfg.postPlugins ?? []),
-    ],
-    globalName: "__lune_temp_global",
-    logLevel: "silent", // we handle errors ourself
-  });
+  if (cfg.builder === "esbuild") {
+    await createEsbuildBuilder(entryPoint, outfile, minify, cfg);
+  } else {
+    await createRolldownBuilder(entryPoint, outfile, minify, root);
+  }
 
   const finalDistJs = (await readFile(outfile)).toString().replace(/var __lune_temp_global\s*=\s*/, "");
```

Comment:
This `___lune_temp_global` is mostly an esbuild workaround, is it also useful for the build output of rolldown? If not it should probably be moved into the esbuild builder

Reply:
I wasn't super sure of what it did, so just kept it as is

Reply:
Yeah essentially the iife esbuild outputs isn't sufficient on its own for dumb esm reasons so we need to tell it to create a global, which makes it actually generate an object, then we just strip off the variable declaration. Its a hack but it works.